akka-projection-slick/src/test/scala/akka/projection/slick/SlickSpec.scala

```diff
@@ -10,16 +10,17 @@ import scala.concurrent.duration._
 import akka.actor.testkit.typed.scaladsl.LogCapturing
 import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
 import akka.projection.slick.internal.SlickOffsetStore
+import akka.projection.slick.internal.SlickSettings
 import akka.projection.testkit.scaladsl.ProjectionTestKit
 import com.typesafe.config.Config
 import slick.basic.DatabaseConfig
 import slick.jdbc.H2Profile
 
 abstract class SlickSpec(config: Config) extends ScalaTestWithActorTestKit(config) with LogCapturing {
 
-  val dbConfig: DatabaseConfig[H2Profile] = DatabaseConfig.forConfig("akka.projection.slick", config)
+  val dbConfig: DatabaseConfig[H2Profile] = DatabaseConfig.forConfig(SlickSettings.configPath, config)
 
-  val offsetStore = new SlickOffsetStore(dbConfig.db, dbConfig.profile)
+  val offsetStore = new SlickOffsetStore(dbConfig.db, dbConfig.profile, SlickSettings(system))
```

Comment:
I find it odd that we have this two kind of config/settings. The reason is that we don't manage the slick database. The user need to create their own `DatabaseConfig`, pass to the projection and make sure they close it when application shuts down. 

Lagom and the JDBC plugin uses a JNDI to create one DB and use it everywhere. I think we should align with the CassandraSessionRegistry on that. That will allow us to guarantee the creation and proper shutdown of the Slick (or JDBC) connection pool (and dedicated thread pool).